File: app/src/main/java/app/fit/fitndflow/domain/usecase/GetSerieAddedUseCase.java
package app.fit.fitndflow.domain.usecase;

import java.util.List;

import app.fit.fitndflow.domain.common.usecase.FitUseCase;
import app.fit.fitndflow.domain.model.GetSerieAddedModel;
import app.fit.fitndflow.domain.model.SerieModel;
import app.fit.fitndflow.domain.repository.FitnFlowRepository;
import io.reactivex.Single;

public class GetSerieAddedUseCase extends FitUseCase<GetSerieAddedModel, List<SerieModel>> {
    private FitnFlowRepository fitnFlowRepository;

    public GetSerieAddedUseCase(int idExercise, String date, FitnFlowRepository fitnFlowRepository){
        super(new GetSerieAddedModel(idExercise, date));
        this.fitnFlowRepository = fitnFlowRepository;
    }
    @Override
    public Single<List<SerieModel>> buildUseCaseObservable() {
        return Single.create(emitter -> {
            try{
                List<SerieModel> response = fitnFlowRepository.getSerieListOfExerciseAdded(inputParams.getDate(), inputParams.getExerciseId());
                emitter.onSuccess(response);
            } catch (Exception e){
                emitter.onError(e);
            }
        });
    }
}

File: domain-layer/src/main/java/app/fit/fitndflow/domain/usecase/GetSerieAddedUseCase.kt
package app.fit.fitndflow.domain.usecase

import app.fit.fitndflow.domain.common.usecase.UseCase
import app.fit.fitndflow.domain.model.SerieModel
import app.fit.fitndflow.domain.repository.FitnFlowRepository
import app.fit.fitndflow.domain.repository.TrainingRepository
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.flow
import javax.inject.Inject

class GetSerieAddedUseCase @Inject constructor(val trainingRepository: TrainingRepository) : UseCase<GetSerieAddedParam, List<SerieModel>>(){
    override fun run(params: GetSerieAddedParam): Flow<List<SerieModel>> = flow {
        val serieAdded = trainingRepository.getSerieListOfExerciseAdded(params.exerciseId)
        emit(serieAdded)
    }
}
data class GetSerieAddedParam(val exerciseId: Int)